src/components/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useEffect } from 'react'
+import { useHistory } from 'react-router-dom'
+import './SignUp.css'
+
+interface PropTypes {
+  form: string;
+  setUserName: any;
+  setUserKey: any;
+}
+
+const SignUp = ({ form, setUserName, setUserKey }: PropTypes) => {
+
+  const [username, setUsername] = useState<string>('')
+  const [password, setPassword] = useState<string>('')
+  const [confirmPassword, setConfirmPassword] = useState<string>('')
+  const [error, setError] = useState<string>('')
+  let history = useHistory()
+
+  useEffect(() => {
+    setError('')
+  }, [form])
+
+  const handleClick = async (e: any) => {
+    e.preventDefault()
+    switch (form) {
+      case 'Log In':
+        try {
+          const params = {
+            "user": {
+              username: username,
+              password: password
+            }
+          }
+          const response = await fetch(`http://localhost:3001/api/v1/login`, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            mode: 'cors',
+            body: JSON.stringify(params)
+          })
+          const data = await response.json();      
+          const apiKey = data.data.attributes.api_key
+          const userName = data.data.attributes.username
+          localStorage.setItem('chessAdventureKey', apiKey)
+          localStorage.setItem('chessAdventureName', userName)
+          setUserName(userName)
+          setUserKey(apiKey)
+          history.push(`/dashboard`)
```

Comment:
In both 'log in' and 'sign up' instances a user has their information sent to local storage and state. Upon returning to the site, if the credentials are there, state is rehydrated with this information when the app loads.